tests/ttnn/unit_tests/test_multi_device_trace.py

```diff
@@ -8,26 +8,31 @@
 import ttnn
 import tempfile
 from loguru import logger
+import os
 from tests.ttnn.utils_for_testing import assert_with_pcc
 from ttnn import ShardTensorToMesh, ReplicateTensorToMesh, ConcatMeshToTensor, ListMeshToTensor
 
+NUM_TRACE_LOOPS = int(os.getenv("NUM_TRACE_LOOPS", 10))
 
-@pytest.mark.parametrize("shape", [(1, 1, 512, 512), (1, 1, 32, 32), (1, 3, 512, 512), (1, 3, 32, 32)])
+
+@pytest.mark.parametrize(
+    "shape", [(1, 1, 512, 512), (1, 1, 32, 32), (1, 3, 32, 32), (1, 1, 256, 256), (1, 3, 512, 512), (1, 3, 128, 128)]
+)
 @pytest.mark.parametrize("use_all_gather", [True, False])
-@pytest.mark.parametrize("enable_async", [True, False])
+@pytest.mark.parametrize("enable_async", [True])
 @pytest.mark.parametrize("device_params", [{"trace_region_size": 33792}], indirect=True)
-def test_multi_device_single_trace(pcie_device_mesh, shape, use_all_gather, enable_async):
-    if pcie_device_mesh.get_num_devices() <= 1:
+def test_multi_device_single_trace(t3k_device_mesh, shape, use_all_gather, enable_async):
+    if t3k_device_mesh.get_num_devices() <= 1:
         pytest.skip("This test requires multiple devices")
 
     # Trace requires program cache to be enabled
-    for device_id in pcie_device_mesh.get_device_ids():
-        pcie_device_mesh.get_device(device_id).enable_async(enable_async)
-        pcie_device_mesh.get_device(device_id).enable_program_cache()
+    for device_id in t3k_device_mesh.get_device_ids():
```

Comment:
btw, you can do `for device in t3k_device_mesh.devices()`

Reply:
Another trace pr coming soon with test. I'll add this change there. Thanks for the suggestion :)